src/mlpack/methods/decision_tree/decision_tree.hpp

```diff
@@ -264,6 +264,86 @@ class DecisionTree :
+   * @param datasetInfo Type information for each dimension of the dataset.
+   * @param labels Labels for each training point.
+   * @param numClasses Number of classes in the dataset.
+   * @param minimumLeafSize Minimum number of points in each leaf node.
+   */
+  template<typename MatType>
+  DecisionTree(MatType& data,
+               const size_t begin,
+               const size_t count,
+               const data::DatasetInfo& datasetInfo,
+               arma::Row<size_t>& labels,
+               const size_t numClasses,
+               const size_t minimumLeafSize = 10);
+
+  /**
+   * Corresponding to the public constructor. 
+   * This method is designed for avoiding unnecessary copies during training.
+   *
+   * @param data Dataset to train on.
+   * @param begin Index of the starting point in the dataset that belongs to this node.
+   * @param count Number of points in this node.
+   * @param labels Labels for each training point.
+   * @param numClasses Number of classes in the dataset.
+   * @param minimumLeafSize Minimum number of points in each leaf node.
+   */
+  template<typename MatType>
+  DecisionTree(MatType& data,
+               const size_t begin,
+               const size_t count,
+               arma::Row<size_t>& labels,
+               const size_t numClasses,
+               const size_t minimumLeafSize = 10);
+
+  /**
+   * Corresponding to the public Train method. 
+   * This method is designed for avoiding unnecessary copies during training.
+   *
+   * @param data Dataset to train on.
+   * @param begin Index of the starting point in the dataset that belongs to this node.
+   * @param count Number of points in this node.
+   * @param datasetInfo Type information for each dimension.
+   * @param labels Labels for each training point.
+   * @param numClasses Number of classes in the dataset.
+   * @param minimumLeafSize Minimum number of points in each leaf node.
+   */
+  template<typename MatType>
+  void Train(MatType& data,
+             const size_t begin,
+             const size_t count,
+             const data::DatasetInfo& datasetInfo,
+             arma::Row<size_t>& labels,
+             const size_t numClasses,
+             const size_t minimumLeafSize = 10);
+
+  /**
+   * Corresponding to the public Train method. 
+   * This method is designed for avoiding unnecessary copies during training.
+   *
+   * @param data Dataset to train on.
+   * @param begin Index of the starting point in the dataset that belongs to this node.
+   * @param count Number of points in this node.
+   * @param labels Labels for each training point.
+   * @param numClasses Number of classes in the dataset.
+   * @param minimumLeafSize Minimum number of points in each leaf node.
+   */
+  template<typename MatType>
+  void Train(MatType& data,
+             const size_t begin,
+             const size_t count,
+             arma::Row<size_t>& labels,
+             const size_t numClasses,
```

Comment:
I placed them under the "private:" at line 236, so I think there is no need to mark them again?